python/mlcroissant/mlcroissant/_src/structure_graph/nodes/field.py

```diff
@@ -130,6 +153,11 @@ def __post_init__(self):
         self.assert_has_mandatory_properties(uuid_field)
         self.source.check_source(self.add_error)
         self._standardize_data_types()
+        if self.array_shape and not self.is_array:
+            self.add_error(
+                f"Field {self.uuid} defines `array_shape`, but `is_array` is False."
+            )
+        self.array_shape_tuple
```

Comment:
Remove?

Reply:
ah you probably use this to check that the string is correct (ints separated by `,`)!

Reply:
Note that this will fail - and not dispatch an error as the other checks.

Reply:
Added explicit error and added one more e2e test.